include/sempr/processing/CallbackModule.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef SEMPR_PROCESSING_CALLBACK_MODULE_HPP_
+#define SEMPR_PROCESSING_CALLBACK_MODULE_HPP_
+
+
+#include "Module.hpp"
+#include "../core/Observable.hpp"
+#include "../core/Utility.hpp"
+#include <functional>
+
+namespace sempr {
+namespace processing {
+
+
+/**
+    The CallbackModule is a convenience module to easily allow adding simple callback functions
+    to react to changes on one specific type of entity.
+    Instantiate it with a callback, e.g. a lambda function, which takes a shared_ptr to an
+    ObservableType as input, e.g. an EntityEvent<...>.
+*/
+template <class ObservableType>
+class CallbackModule : public Module<ObservableType> {
+public:
+    typedef std::function<void(std::shared_ptr<ObservableType>)> callback_t;
```

Comment:
Strange naming. *_t is the default struct naming of the C Std Lib. So this confused me at the first look.

Reply:
What name / naming convention would you propose for such typedefs? I'm always struggling to find good names... :wink:

Reply:
Mhhh good question.
Mostly I use camel-case with an upper letter at start for classes and types. A lower start letter for methods.
But this is a function typedef, so its more difficult.

I suggest something like "callbackType" will fit well.

Reply:
Hmmm, for some (subjective) reason I don't really like that. Maybe because I mostly interpret camel-case names starting with a lowercase letter as local variables? Also, explicitly having "Type" in the name of a type seems strange. Though I have to admit that the "_t" suffix is more or less the same - but shorter. :wink:

Reply:
But the "_t" suffix seems like the "oldschool" C types e.g. "size_t", "clock_t", "wchar_t", "time_t".
This trigger nervous goosebumps in my case ;-)

What about something different like "observable_cb"